app/server.go

```diff
@@ -781,6 +791,46 @@ func doSessionCleanup(s *Server) {
 	s.Store.Session().Cleanup(model.GetMillis(), SESSIONS_CLEANUP_BATCH_SIZE)
 }
 
+func doLicenseExpirationCheck(s *Server) {
+	s.FakeApp().LoadLicense()
+	license := s.License()
+
+	if license == nil {
+		mlog.Info("License cannot be found.")
+		return
+	}
+
+	if !license.IsPastGracePeriod() {
+		mlog.Info("License is not past the grace period.")
+		return
+	}
+
+	users, err := s.Store.User().GetSystemAdminProfiles()
+	if err != nil {
+		mlog.Error("Failed to get system admins for license expired message from Mattermost.")
+		return
+	}
+
+	//send email to admin(s)
+	for _, user := range users {
+		user := user
+		if user.Email == "" {
+			mlog.Error("Invalid system admin email.", mlog.String("user_email", user.Email))
+			continue
+		}
+
+		mlog.Info("Sending license expired email.", mlog.String("user_email", user.Email))
```

Comment:
If we warn about the fact that we are removing the license and we have notified the admins, probably not showing this log message is perfectly fine.

Reply:
The log entry is for us and is for a different operation (sending email) - the admin notification is for the user; same as above, we can remove it or make it Debug. Thoughts?

Reply:
Yes, I think Debug would be more appropriate, but if you move the log of "RemoveLicense" to the "RemoveLicense" method, then, that log message will be generic, and then, probably makes sense to log it here.

Reply:
yes, I was thinking of moving  it to RemoveLicense() as well, it makes more sense.